tests/unittests/test_request_timeout.py

```diff
@@ -0,0 +1,396 @@
+import unittest
+from unittest import mock
+
+from requests.exceptions import Timeout
+import tap_bing_ads
+from urllib.error import URLError
+from tap_bing_ads import CustomServiceClient
+import datetime
+    
+class MockClient():
+    '''Mocked ServiceClient class and it's method to pass the test case'''
+    def __init__(self, error):
+        self.error = error
+        self.count = 0
+
+    def GetCampaignsByAccountId(self, AccountId, CampaignType):
+        '''Mocked GetCampaignsByAccountId method of the Client class'''
+        self.count = self.count + 1
+        raise self.error
+
+    def GetAdGroupsByCampaignId(self, CampaignId):
+        '''Mocked GetAdGroupsByCampaignId method of the Client class'''
+        self.count = self.count + 1
+        raise self.error
+
+    def GetAdsByAdGroupId(self, AdGroupId, AdTypes):
+        '''Mocked GetAdsByAdGroupId method of the Client class'''
+        self.count = self.count + 1
+        raise self.error
+
+    def PollGenerateReport(self):
+        '''Mocked PollGenerateReport method of the Client class'''
+        self.count = self.count + 1
+        raise self.error
+
+    def SubmitGenerateReport(self, report_request):
+        '''Mocked SubmitGenerateReport method of the Client class'''
+        self.count = self.count + 1
+        raise self.error
+
+    def PollGenerateReport(self, request_id):
+        '''Mocked PollGenerateReport method of the Client class'''
+        self.count = self.count + 1
+        raise self.error
+
+    @property
+    def factory(self):
+        '''Mocked factory property of the Client class'''
+        self.count = self.count + 1
+        raise self.error
+
+    @property
+    def soap_client(self):
+        '''Mocked soap_client property of the Client class'''
+        self.count = self.count + 1
+        raise self.error
+
+    @property
+    def call_count(self):
+        '''Mocked call_count property of the Client class'''
+        return self.count
+
+@mock.patch("tap_bing_ads.filter_selected_fields_many", return_value = '')
+@mock.patch("singer.write_records", return_value = '')
+@mock.patch("singer.metrics", return_value = '')
+@mock.patch("singer.write_bookmark", return_value = '')
+@mock.patch("singer.write_state", return_value = '')
+@mock.patch("tap_bing_ads.sobject_to_dict", return_value = '')
+@mock.patch("singer.get_bookmark", return_value = {'b1': 'b1'})
+@mock.patch("singer.write_schema", return_value = '')
+@mock.patch("tap_bing_ads.get_core_schema", return_value = '')
+@mock.patch("tap_bing_ads.get_selected_fields", return_value = '')
+class TestBackoffError(unittest.TestCase):
```

Comment:
@namrata270998 What is the relevance of these test cases in this PR?
CC: @dbshah1212

Reply:
These backoff checks that the request is backoff properly when it encounters URLError with time out message

Reply:
But how are they relevant with your code changes in this PR?